File: Greedy/python/1744.py
n = int(input())
minus_zero = []
plus = []
result = 0
for _ in range(n):
  num = int(input())
  if num<=0:
    minus_zero.append(num)
    continue
  if num==1:
    result+=1
    continue
  plus.append(num)
minus_zero.sort(reverse=True)
plus.sort()
while len(minus_zero)>1:
  result += minus_zero.pop() * minus_zero.pop()
while len(plus)>1:
  result += plus.pop() * plus.pop()
if len(minus_zero)>0:
  result += minus_zero[-1]
if len(plus)>0:
  result += plus[-1]
print(result)

File: Greedy/javascript/1744.js
const input = require("fs").readFileSync("/dev/stdin").toString().split("\n");

const minusAndZero = [];
const plusNum = [];
let result = 0;
for (let i = 1; i < input.length - 1; i++) {
  if (input[i] <= 0) minusAndZero.push(Number(input[i]));
  else if (input[i] == 1) result++;
  else plusNum.push(Number(input[i]));
}
minusAndZero.sort((a, b) => b - a);
plusNum.sort((a, b) => a - b);

while (minusAndZero.length > 1) {
  result += minusAndZero.pop() * minusAndZero.pop();
}
if (minusAndZero.length > 0) result += minusAndZero.pop();

while (plusNum.length > 1) {
  result += plusNum.pop() * plusNum.pop();
}
if (plusNum.length > 0) result += plusNum.pop();

console.log(result);
